core/drivers/firewall/stm32_etzpc.c

```diff
@@ -372,6 +409,211 @@ stm32_etzpc_acquire_memory_access(struct firewall_query *firewall,
+					 enum etzpc_decprot_attributes attr)
+{
+	paddr_t ram_base = 0;
+	size_t ram_size = 0;
+
+	if (!IS_ENABLED(CFG_WITH_PAGER))
+		return true;
+
+	switch (decprot_id) {
+	case ETZPC_TZMA1_ID:
+		ram_base = SYSRAM_BASE;
+		ram_size = SYSRAM_SEC_SIZE;
+		break;
+	case STM32MP1_ETZPC_SRAM1_ID:
+		ram_base = SRAM1_BASE;
+		ram_size = SRAM1_SIZE;
+		break;
+	case STM32MP1_ETZPC_SRAM2_ID:
+		ram_base = SRAM2_BASE;
+		ram_size = SRAM2_SIZE;
+		break;
+	case STM32MP1_ETZPC_SRAM3_ID:
+		ram_base = SRAM3_BASE;
+		ram_size = SRAM3_SIZE;
+		break;
+	case STM32MP1_ETZPC_SRAM4_ID:
+		ram_base = SRAM4_BASE;
+		ram_size = SRAM4_SIZE;
+		break;
+	default:
+		return true;
+	}
+
+	if (stm32mp1_ram_intersect_pager_ram(ram_base, ram_size) &&
+	    attr != ETZPC_DECPROT_S_RW) {
+		EMSG("Internal RAM %#"PRIxPA"..%#"PRIxPA" is used by pager, shall be secure",
+		     ram_base, ram_base + ram_size);
+		return false;
+	}
+
+	return true;
+}
+#endif /* CFG_STM32MP15 */
+
+static bool decprot_id_is_sram(uint32_t id)
+{
+	switch (id) {
+	case STM32MP1_ETZPC_SRAM1_ID:
+	case STM32MP1_ETZPC_SRAM2_ID:
+	case STM32MP1_ETZPC_SRAM3_ID:
+#ifdef CFG_STM32MP15
+	case STM32MP1_ETZPC_SRAM4_ID:
+	case STM32MP1_ETZPC_RETRAM_ID:
+#endif
+		return true;
+	default:
+		return false;
+	}
+}
+
+static TEE_Result stm32_etzpc_configure_memory(struct firewall_query *firewall,
+					       paddr_t paddr, size_t size)
+{
+	enum etzpc_decprot_attributes attr = ETZPC_DECPROT_MAX;
+	bool lock = false;
+	uint32_t mode = 0;
+	uint32_t id = 0;
+
+	if (firewall->arg_count != 1)
+		return TEE_ERROR_BAD_PARAMETERS;
+
+	id = firewall->args[0] & ETZPC_ID_MASK;
+	mode = (firewall->args[0] & ETZPC_MODE_MASK) >> ETZPC_MODE_SHIFT;
+	attr = etzpc_binding2decprot(mode);
+	lock = firewall->args[0] & ETZPC_LOCK_MASK;
+
+	if (decprot_id_is_sram(id)) {
+		/* Use OP-TEE SRAM addresses, not the alias one */
+		paddr = stm32mp1_pa_or_sram_alias_pa(paddr);
+
+		/* Target address range must match the full SRAM range */
+		switch (id) {
+		case STM32MP1_ETZPC_SRAM1_ID:
+			if (paddr != SRAM1_BASE || size != SRAM1_SIZE)
+				return TEE_ERROR_BAD_PARAMETERS;
+			break;
+		case STM32MP1_ETZPC_SRAM2_ID:
+			if (paddr != SRAM2_BASE || size != SRAM2_SIZE)
+				return TEE_ERROR_BAD_PARAMETERS;
+			break;
+		case STM32MP1_ETZPC_SRAM3_ID:
+			if (paddr != SRAM3_BASE || size != SRAM3_SIZE)
+				return TEE_ERROR_BAD_PARAMETERS;
+			break;
+#if defined(CFG_STM32MP15)
+		case STM32MP1_ETZPC_SRAM4_ID:
+			if (paddr != SRAM4_BASE || size != SRAM4_SIZE)
+				return TEE_ERROR_BAD_PARAMETERS;
+			break;
+		case STM32MP1_ETZPC_RETRAM_ID:
+			if (paddr != RETRAM_BASE || size != RETRAM_SIZE)
+				return TEE_ERROR_BAD_PARAMETERS;
+			break;
+#endif /* defined(CFG_STM32MP15) */
+		default:
+			panic();
+		}
+
+		if (decprot_is_locked(id)) {
+			if (etzpc_get_decprot(id) != attr) {
+				EMSG("Internal RAM configuration locked");
+				return TEE_ERROR_ACCESS_DENIED;
+			}
+
+			sanitize_decprot_config(id, attr);
```

Comment:
Not needed as the decprot is locked so nothing is changed

Reply:
What is the locked configuration does is not compliant, according to sanitize?
I prefer to have a check here, e.g. SRAM1 locked mckprot but RCC mckprot disabled.

Reply:
There can't be a need to sanitize where the config is already locked IMO. We've made no change, and nothing can be done